src/common/ExpenseClasses/ExpenseClasses.js

```diff
@@ -15,18 +15,33 @@ const columnMapping = {
   percentageExpended: <FormattedMessage id="ui-finance.budget.expenseClasses.percentageExpended" />,
   status: <FormattedMessage id="ui-finance.budget.expenseClasses.status" />,
 };
-const resultsFormatter = {
-  encumbered: expenseClass => <AmountWithCurrencyField amount={expenseClass.encumbered} />,
-  awaitingPayment: expenseClass => <AmountWithCurrencyField amount={expenseClass.awaitingPayment} />,
-  expended: expenseClass => <AmountWithCurrencyField amount={expenseClass.expended} />,
-  percentageExpended: expenseClass => (isNumber(expenseClass.percentageExpended)
-    ? `${expenseClass.percentageExpended}%`
-    : <FormattedMessage id="ui-finance.budget.expenseClasses.percentageExpended.undefined" />
-  ),
-  status: expenseClass => <FormattedMessage id={`ui-finance.budget.expenseClasses.status.${expenseClass.expenseClassStatus}`} />,
-};
+const ExpenseClasses = ({ currency, expenseClassesTotals, visibleColumns, id }) => {
+  const resultsFormatter = useMemo(() => ({
```

Comment:
No need to put it under component, leave it as is, just make it as a function passing currency